### ISSUE TITLE
bug in validation during training

### ISSUE THREAD

**Issue by wyli** [NifTK/NiftyNet#28]
For the 'validation during training'feature:

When enabled, it should compute gradients and update the network using data from the training set, and compute losses on the validation set after every N training iterations.

But in the current version, at each validation step, the network gets
updated using gradients computed from validation data.

As a result, it trains with validation set data unexpectedly and gives
wrong validation curves.

also filed at https://cmiclab.cs.ucl.ac.uk/CMIC/NiftyNet/issues/214
